packages/engine/Source/Scene/ITwinData.js

```diff
@@ -0,0 +1,79 @@
+import Cesium3DTileset from "./Cesium3DTileset.js";
+import defined from "../Core/defined.js";
+import Resource from "../Core/Resource.js";
+import ITwinPlatform from "../Core/ITwinPlatform.js";
+import RuntimeError from "../Core/RuntimeError.js";
+import Check from "../Core/Check.js";
+
+/**
+ * Methods for loading iTwin platform data into CesiumJS
+ *
+ * @experimental This feature is not final and is subject to change without Cesium's standard deprecation policy.
+ *
+ * @see ITwinPlatform to set the API token and base api url
+ * @namespace ITwinData
+ */
+const ITwinData = {};
+
+/**
+ * @param {ExportRepresentation} exportObj
+ * @param {Cesium3DTileset.ConstructorOptions} [options] Object containing options to pass to an internally created {@link Cesium3DTileset}.
+ * @returns {Promise<Cesium3DTileset>}
+ */
+async function loadExport(exportObj, options) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.defined("exportObj", exportObj);
+  //>>includeEnd('debug')
+
+  if (exportObj.request.exportType !== ITwinPlatform.ExportType["3DTILES"]) {
+    throw new RuntimeError(`Wrong export type ${exportObj.request.exportType}`);
+  }
+  if (exportObj.status !== ITwinPlatform.ExportStatus.Complete) {
+    throw new RuntimeError(
+      `Export is not completed. ${exportObj.id} - ${exportObj.status}`,
+    );
+  }
+
+  // Convert the link to the tileset url while preserving the search paramaters
+  // This link is only valid 1 hour
+  const baseUrl = new URL(exportObj._links.mesh.href);
+  baseUrl.pathname = `${baseUrl.pathname}/tileset.json`;
+  const tilesetUrl = baseUrl.toString();
+
+  const resource = new Resource({
+    url: tilesetUrl,
+  });
+
+  return Cesium3DTileset.fromUrl(resource, options);
+}
+
+/**
+ * Loads the export for the specified iModel with the export type that CesiumJS can load and returns
+ * a tileset created from that export.
+ *
+ * If the export is not finished processing this will throw an error. It is up to the caller
+ * to re-attempt loading at a later time
+ *
+ * @example
+ * const tileset = await Cesium.ITwinData.createTilesetFromModelId(imodelId);
+ * viewer.scene.primitives.add(tileset);
+ *
+ * @experimental This feature is not final and is subject to change without Cesium's standard deprecation policy.
+ *
+ * @param {string} iModelId The id of the iModel to load
+ * @param {Cesium3DTileset.ConstructorOptions} [options] Object containing options to pass to an internally created {@link Cesium3DTileset}.
+ * @returns {Promise<Cesium3DTileset | undefined>} Will return <code>undefined</code> if there is no export for the given iModel id
+ *
+ * @throws {RuntimeError} Wrong export type [type]
+ * @throws {RuntimeError} Export is not completed. [id] - [status]
+ */
+ITwinData.createTilesetFromModelId = async function (iModelId, options) {
+  const { exports } = await ITwinPlatform.getExports(iModelId);
```

Comment:
If I recall correctly, I think we discussed with @danieliborra that we should be caching the results of this call on the client to avoid overloading the REST API.

We do this occasionally in other parts of the API. For instance, see `requestMetadata` in `BingMapsImageryProvider.js`.

Reply:
I believe caching was suggested only if the Sandcastle was going to get a massive amount of traffic?

Regardless of that part I don't think we want to cache this without some extra thought. This route can return different results for the same input. For example when:
- The export is still processing and not `status = Complete`
- When the iModel has changed and needs to have a new export for the latest changeset

Reply:
From the numbers Gabby provided, we won't have problems, so you can skip the cache, it will avoid headaches.
I was only afraid in the case where we had hundreds of concurrent users, which won't be the case (get-exports returns in 1-2 seconds max, so 1000 concurrent users would mean 1M/h users, which I assume won't happen).